src/Evolution/DiscontinuousGalerkin/Filtering.hpp

```diff
@@ -0,0 +1,281 @@
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace dg {
+namespace Actions {
+/// \cond
+template <size_t FilterIndex, typename TagsToFilterList>
+struct ExponentialFilter;
+/// \endcond
+
+/*!
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief Applies an exponential filter to the specified tags.
+ *
+ * Applies an exponential filter in each logical direction to each component of
+ * the tensors `TagsToFilter`. The exponential filter rescales the 1d modal
+ * coefficients \f$c_i\f$ as:
+ *
+ * \f{align*}{
+ *  c_i\to c_i \exp\left[-\alpha_{\mathrm{ef}}
+ *   \left(\frac{i}{N}\right)^{2\beta_{\mathrm{ef}}}\right]
+ * \f}
+ *
+ * where \f$N\f$ is the basis order, \f$\alpha_{\mathrm{ef}}\f$ determines how
+ * much the coefficients are rescaled, and \f$\beta_{\mathrm{ef}}\f$ (given by
+ * the `HalfPower` option) determines how aggressive/broad the filter is (lower
+ * values means filtering more coefficients). Setting
+ * \f$\alpha_{\mathrm{ef}}=36\f$ results in effectively zeroing the highest
+ * coefficient (in practice it gets rescaled by machine epsilon). The same
+ * \f$\alpha_{\mathrm{ef}}\f$ and \f$\beta_{\mathrm{ef}}\f$ are used in each
+ * logical direction.
+ *
+ * If different `Alpha` or `HalfPower` parameters are desired for different tags
+ * then multiple `ExponentialFilter` actions must be inserted into the action
+ * list with different `FilterIndex` values. In the input file these will be
+ * specified as `ExpFilterFILTER_INDEX`, e.g. `ExpFilter0` and `ExpFilter1`.
+ * Below is an example of an action list with two different exponential filters:
+ *
+ * \snippet DiscontinuousGalerkin/Test_Filtering.cpp action_list_example
+ *
+ * #### Action properties
+ *
+ * Uses:
+ * - ConstGlobalCache:
+ *   - ExponentialFilter
+ * - DataBox:
+ *   - Tags::Mesh
+ * DataBox changes:
+ * - Adds: nothing
+ * - Removes: nothing
+ * - Modifies:
+ *   - TagsToFilter
+ *
+ * #### Design decision:
+ *
+ * - The reason for the `size_t` template parameter is to allow for different
+ * `Alpha` and `HalfPower` parameters for different tensors while still being
+ * able to cache the matrices.
+ */
+template <size_t FilterIndex, typename... TagsToFilter>
+class ExponentialFilter<FilterIndex, tmpl::list<TagsToFilter...>> {
+ public:
+  using const_global_cache_tags = tmpl::list<ExponentialFilter>;
+
+  /// \brief The value of `exp(-alpha)` is what the highest modal coefficient is
+  /// rescaled by.
+  struct Alpha {
+    using type = double;
+    static constexpr OptionString help =
+        "exp(-alpha) is rescaling of highest coefficient";
+  };
+  /*!
+   * \brief Half of the exponent in the exponential.
+   *
+   * \f{align*}{
+   *  c_i\to c_i \exp\left[-\alpha \left(\frac{i}{N}\right)^{2m}\right]
+   * \f}
+   */
+  struct HalfPower {
+    using type = unsigned;
+    static constexpr OptionString help =
+        "Half of the exponent in the generalized Gaussian";
+  };
```

Comment:
I'm not sure what you're suggesting be changed. Things I can think of are:
1) The name of the option
2) That it be the power instead of half power
3) Not allowing people to set this option
4) Something else?

My responses to these are:
1) I'm open to suggestions, but I think the current one is very clear.
2) Using the power is wrong because the power needs to be even. Setting the half power prevents a user from ever getting it to be odd without having to check anything and is consistent with Hesthaven.
3) Since this controls how many modes are filtered I think it's a necessary option.
4) I'd need clarification to respond.

Reply:
I was concerned about point 2. Thanks.